include/exadg/operators/inverse_mass_operator.h

```diff
@@ -91,9 +109,36 @@ class InverseMassOperator
     }
   }
 
+  void
+  initialize_inverse_mass_operator_with_block_jacobi()
+  {
+    // initialize mass operator
+    dealii::AffineConstraints<Number> const & constraint =
+      matrix_free->get_affine_constraints(dof_index);
+
+    MassOperatorData<dim> mass_operator_data;
+    mass_operator_data.dof_index  = dof_index;
+    mass_operator_data.quad_index = quad_index;
+
+    mass_operator.initialize(*matrix_free, constraint, mass_operator_data);
```

Comment:
Why creating a new mass operator? Doesn't it already exist.

The variable `implement_block_diagonal_preconditioner_matrix_free` in operator data is not set here. Hence, I do not see how the present implementation of the inverse mass operator can support the matrix-free variant with elementwise iterative solver.

Note also that the solver tolerances in operator data are by default very coarse (reltol = 1e-2), since the block-jacobi preconditioner is typically used as a preconditioner. Here, however, the block Jacobi preconditioner is used as a solver, i.e. to invert the mass operator. With the default setting, I expect that spatial errors and convergence rates for simplices are severly limited, because solver tolerances for the inverse mass operator dominate over all other errors of the spatial discretization for sufficiently fine spatial resolutions.

Having said this, it could indeed make sense to create a new mass operator here, but the current implementation seems to be incomplete. Did you test the different variants?

Reply:
This makes me think about the original approach I had, because passing `implement_block_diagonal_preconditioner_matrix_free` doesn't seem easy. Maybe we need to construct the mass operator in spatial_operator_base and pass over to inverse_mass_operator, by extending the interface. But this would result in if-else branches in higher levels. This was the argument against this approach.

Reply:
As I said above, it can make sense to create a new mass operator here, I am not per se against it but want to discuss it. With the arguments you provided (velocity vs. pressure), I think this is a reasonable decision for now.